### PR TITLE
JSON API: Add maybe auto-update-api-endpoint

### PR DESCRIPTION
**Pull request by enejb**
Add a new Maybe Auto Update API Endpoint that will allows us to trigger any autoupdates right away. 
As opposed to wait for the scheduled time at which they are happening.

### CODE REVIEW

**Comment by samhotchkiss** [comment]
LGTM!

**Comment by enejb** [comment]
When everything goes well we get this log 
screen shot 2015-11-17 at 11 02 12 am

When the site is out of space we get the following.
screen shot 2015-11-17 at 11 59 50 am

There is no log if the site is under version control or auto updates are disabled and the update fails.
screen shot 2015-11-17 at 11 06 26 am

**Comment by enejb** [comment]
Updated to include reasons why the update would fail. 
screen shot 2015-11-17 at 3 27 34 pm

File: json-endpoints/jetpack/class.jetpack-json-api-maybe-auto-update-endpoint.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+class Jetpack_JSON_API_Maybe_Auto_Update_Endpoint extends Jetpack_JSON_API_Endpoint {
+	// POST /sites/%s/maybe_auto_update
+	protected $needed_capabilities = array( 'update_core', 'update_plugins', 'update_themes' );
+
+	protected $update_results = array();
+
+	protected function result() {
+		add_action( 'automatic_updates_complete', array( $this, 'get_update_results' ), 100, 1 );
+
+		include_once( ABSPATH . '/wp-admin/includes/admin.php' );
```
Comment (lezama) [inline json-endpoints/jetpack/class.jetpack-json-api-maybe-auto-update-endpoint.php]: we can call `wp_maybe_auto_update()` directly https://github.com/WordPress/WordPress/blob/77e365efbf2e499e2ed11d29c101ea466cf1ceed/wp-includes/update.php#L493

**Comment by enejb** [comment]
Move the errors to an error array. Which I am hoping to keep the response a bit cleaner.
screen shot 2015-11-19 at 1 21 28 pm

**Comment by roccotripaldi** [comment]
This looks good and works well!

**Pull request merged**